### ISSUE TITLE
Add features for assurance levels

### ISSUE THREAD

**Issue by franziskuskiefer** [cryspen/libcrux#314]
Add features for the different assurance levels in libcrux and it's sub-crates.

- Correctness proof
- Panic free
- unverified/experimental

**Comment by franziskuskiefer** [comment]
* pure rust

**Comment by franziskuskiefer** [comment]
@jschneider-bensch is adding some features in #329. Can you update what the plan is here?

**Comment by franziskuskiefer** [comment]
There's some work happening in #448. @jschneider-bensch can we get that PR and either close this issue or describe what else we should do here for now?

**Comment by jschneider-bensch** [comment]
#448 should not introduce new features, only describe the verification status in each crate. The idea is to delineate verified algorithms (e.g. the code we import from HACL) from verified APIs (e.g. a thin Rust API on top of verified HACL code). This way we can be a bit more nuanced in our description of verification status, as opposed to a minimal binary verification status.
In relation to this issue: 
- I don't think it makes sense to give features for the API/algorithm distinction, since in most cases the verified algorithms will only be accessible through (an otherwise empty) as yet unverified API. 
- We have the `pre-verification` feature in `libcrux-ml-kem` to distinguish verified from as yet unverified code. We could, for now continue this split across the crates in the workspace.

**Comment by franziskuskiefer** [comment]
Yeah I meant that #448 adds some needed documentation for these levels. Let's add issues in here for the crates where we want to add features for this.

**Comment by jschneider-bensch** [comment]
Okay, at the moment I think the assurance level features make sense only for top-level libcrux and for ML-DSA (at the moment). The individual crates have a certain verification status that should be highlighted (#672, #569), but there we don't have verified and unverified implementations of the same functionality (except for ML-DSA  where you could have a "pre-verification" type feature for the NEON implementation).
